extensions/amp-story/1.0/amp-story-interactive-results.js

```diff
@@ -56,23 +61,66 @@ const buildResultsTemplate = (element) => {
  * Processes the state and returns the condensed results.
  * @param {!Map<string, {option: ?./amp-story-interactive.OptionConfigType, interactiveId: string}>} interactiveState
  * @param {?Array<!./amp-story-interactive.OptionConfigType>} options needed to ensure the first options take precedence on ties
+ * @param {string} strategy
  * @return {InteractiveResultsDef} the results
  */
-const processResults = (interactiveState, options) => {
+const processResults = (interactiveState, options, strategy) => {
```

Comment:
Is there any way to make this code easier to read? Does extracting some of the blocks into other functions make this function easier?

Also, could you add some unit tests like this one? Let's keep each one of them very focused on asserting one behavior so it's easy for someone changing this code to know exactly what they broke, or even understand all the cases that this piece of code handles.

https://github.com/ampproject/amphtml/blob/9e5f0bff9063c49e822dd0f71d84780f04f1fb35/extensions/amp-story/1.0/test/test-amp-story-interactive-results.js#L83-L96